services/QuillLMS/engines/evidence/spec/lib/open_ai/concerns/api_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Evidence
+  RSpec.describe(OpenAI::Concerns::API, type: :model) do
+
+    let(:endpoint) {'https://api.openai.com/v1/some_endpoint'}
+    let(:result_value_url) {'https://test.com'}
+    let(:sample_request) {{'key1' => 'value'}}
+    let(:sample_response_body) do
+      {
+        "key"=>"value",
+      }
+    end
+    # include headers in response for proper parsing by HTTParty
+    let(:sample_response) { {body: sample_response_body.to_json, headers: {content_type: 'application/json'}} }
+
+    let(:class_without_defined_methods) do
```

Comment:
Cool.  I like this pattern for testing for implementation of the concern's methods.